index.bs

```diff
@@ -3781,6 +3810,802 @@ opened</dfn> steps given |window|, |type| and |message|.
+
+#### The network.Initiator type #### {#type-network-Initiator}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.Initiator = {
+    type: "parser" / "script" / "preflight" / "other",
+    ?columnNumber: uint,
+    ?lineNumber: uint,
+    ?stackTrace: StackTrace,
+    ?request: network.Request
+};
+</pre>
+
+The <code>network.Initiatior</code> type represents the source of a network
+request.
+
+<div algorithm>
+To <dfn>get the initiator</dfn> given |request|:
+
+1. Let |request id| be |request|'s [=request id=].
+
+1. Let |type| be "<code>other</code>".
+
+1. If |request| is a [=CORS-Preflight Request=], set |type| to
+   "<code>preflight</code>".
+
+1. TODO: Get the |type|. It's not quite clear how this ought to work; the CDP
+   data depends on whether the navigation was kicked off by the parser or by
+   script (so e.g. inserting an image from script causes the initiator to be
+   "<code>script</code>"), but that doesn't correspond to anything in Fetch.
+
+1. If |request|'s [=request/initiator type=] is "<code>fetch</code>" or
+   "<code>xmlhttprequest</code>":
+
+  1. Let |stack trace| be the [=current stack trace=].
+
+  1. If |stack trace| has size of 1 or greater, let |line number| be value of the
+     <code>lineNumber</code> field in |stack trace|[0], and let |column number| be
+     the value of the <code>columnNumber</code> field |stack trace|[0]. Otherwise
+     let |line number| and |column number| be 0.
+
+  Otherwise, let |stack trace|, |column number|, and |line number| all be null.
+
+  TODO: Chrome includes the current parser position as column number / line
+  number for parser-inserted resources.
+
+1. Return a [=map=] matching the <code>network.Initiator</code> production, with
+   the <code>type</code> field set to |type|, the <code>columnNumber</code>
+   field set to |column number| if it's not null, or omitted otherwise, the
+   <code>lineNumber</code> field set to |line number| if it's not null, or
+   omitted otherwise, the <code>stackTrace</code> field set to |stack trace| if
+   it's not null, or omitted otherwise, and the <code>request</code> field set
+   to |request id|.
+
+</div>
+
+#### The network.Request type #### {#type-network-Request}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+network.Request = text;
+</pre>
+
+Each network request has an associated <dfn export>request id</dfn>, which is a
+string uniquely identifying that request. The identifier for a request resulting from a
+redirect matches that of the request that initiated it.
+
+#### The network.RequestData type #### {#type-network-RequestData}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.RequestData = {
+    request: network.Request,
+    url: text,
+    method: text,
+    headers: [*network.Header],
+    cookies: [*network.Cookie],
+    headersSize: uint,
+    bodySize: uint / null,
```

Comment:
Based on the steps below, bodySize is either 0 or body's length, so it can't be null?